web/src/components/vis/mixins/axisPlot.js

```diff
@@ -11,46 +10,72 @@ function labelAxis(label, msg, xFunc, yFunc, rot) {
 }
 
 export const axisPlot = {
-  methods: {
-    axisPlot({
-      margin, xrange, yrange, xlabel, ylabel, duration,
-    }) {
-      // Collect necessary props.
-      const {
-        width,
-        height,
-      } = this.$props;
-
-      // Grab the root SVG element.
-      const svg = select(this.$refs.svg);
-      this.svg = svg;
-
-      // Compute the size of the data rectangle.
-      this.margin = margin;
-
-      const dwidth = width - margin.left - margin.right;
-      this.dwidth = dwidth;
-
-      const dheight = height - margin.top - margin.bottom;
-      this.dheight = dheight;
-
-      // Create X and Y scales.
-      const scaleX = scaleLinear()
+  props: {
+    width: {
+      type: Number,
+      required: true,
+    },
+    height: {
+      type: Number,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      axisPlotInitialized: false,
+      margin: {
+        top: 20,
+        right: 0,
+        bottom: 50,
+        left: 50,
+      },
```

Comment:
`margin` is passed into `axisPlot()` below as a required statement from the vis component, so perhaps this initializer should contain `null`s instead of these numbers?

Reply:
(possibly the same question for `xrange` and `yrange` below, though it makes sense to supply reasonable defaults for `xlabel`, `ylabel`, and `duration` IMO--even `xrange` and `yrange` might have reasonable defaults, but likely not `[-1.2, 1.2]`)

Reply:
I made things `null` that are required to be sent from the base component, and put in reasonable defaults for the rest.